src/components/App.js

```diff
@@ -1,15 +1,73 @@
 /* eslint-disable react-hooks/exhaustive-deps */
 import { useQuery } from '@apollo/client';
 import { GET_ALL_ITEMS } from '../api';
+import { Route, Routes, NavLink } from 'react-router-dom';
 import { useState, useEffect } from 'react';
-import Landing from './Landing';
-import Products from './Products';
-import About from './About';
+import ShoppingBag from './ShoppingBag';
+import Checkout from './Checkout';
+import Home from './Home';
+
 
 function App() {
+  const [shoppingBag, setShoppingBag] = useState(
+    [
+      {
+        id: 1,  
+        type: "bracelet",
+        color: "moss",
+        size: 'M',
+        quantity: 2,
+        price: 10.00,
+        image: 'https://cdn.shopify.com/s/files/1/0192/8012/products/friendship-bracelet-adjustable-camp-minimalist-rope-dowling-brothers-bangle-jewellery-740.jpg'
+      },
+      {
+        id: 2,
+        type: "bracelet",
+        color: "orange plaid",
+        size: 'S',
+        quantity: 3,
+        price: 10.00,
+        image: 'https://cdn.shopify.com/s/files/1/0192/8012/products/friendship-bracelet-adjustable-camp-minimalist-rope-dowling-brothers-bangle-jewellery-740.jpg'
+      },
+      {
+        id: 3,
+        type: "leash",
+        color: "lime",
+        size: 'onesize',
+        quantity: 1,
+        price: 20.00,
+        image: 'https://cdn.shopify.com/s/files/1/0192/8012/products/friendship-bracelet-adjustable-camp-minimalist-rope-dowling-brothers-bangle-jewellery-740.jpg'
+      }       
+  ])
+  const [totalPrice, setTotalPrice] = useState(0);
   const [items, setItems] = useState([])
   const [itemsForDisplay, setItemsForDisplay] = useState([])
 
+  const addTotalPrice = () => {
+    const total = shoppingBag.reduce((price, item) => {
+      return price += (item.price * item.quantity)
+    }, 0)
+    setTotalPrice(total)
+  }
+
+  useEffect(() => {
+    addTotalPrice();
+  }, [shoppingBag])
+
+  const removeItemFromBag = id => {
+    setShoppingBag(shoppingBag.filter(item => item.id !== parseInt(id)))
+  }
+
+  const updateQuantity = (id, operation = 'subtract') => {
```

Comment:
oh cool, never thought to use default param like that

Reply:
I had heard of it but, but looked it up so I could use it this time. Fun stuff!